base/headers.go

```diff
@@ -12,6 +12,27 @@ import "strings"
 // Whitespace recognised by SIP protocol.
 const c_ABNF_WS = " \t"
 
+// Maybestring contains a string, or nil.
+type MaybeString interface {
+	implementsMaybeString()
+}
+
+// NoString represents the absence of a string.
+type NoString struct{}
+
+func (n NoString) implementsMaybeString() {}
+
+// String represents an actual string.
+type String struct {
+	S string
```

Comment:
I think this should be private. Callers can get the string with String(), and should set the string by creating a new MaybeString struct; MaybeString should feel immutable in the same way as the built-in 'string'.